rr_rviz_plugins/src/StartlightPanel.cpp

```diff
@@ -0,0 +1,96 @@
+//
+// Created by nico on 11/3/19.
+//
+#include <pluginlib/class_list_macros.h>
+#include <rr_rviz_plugins/StartlightPanel.h>
+#include <std_msgs/Bool.h>
+#include <QVBoxLayout>
+#include <QtGui/QPaintEvent>
+#include <QtGui/QPainter>
+#include <QtWidgets/QLabel>
+
+namespace rr_rviz_plugins {
+bool isGreen = false;
+bool receivedSignal = false;
+QLabel *label;
+ros::Time current;
+ros::Time callbackTime = ros::Time::now();
+ros::Timer timer;
+StartlightPanel::StartlightPanel(QWidget *parent)
+      : rviz::Panel(parent)  // Base class constructor
+{
+    auto *layout = new QVBoxLayout;
+    label = new QLabel("              No Message");
+    label->setGeometry(QRect(80, 0, 30, 30));
+    start_detector = nh.subscribe<std_msgs::Bool>("/start_detected", 1, StartlightPanel::startlightCallback);
+    timer = nh.createTimer(ros::Duration(1), StartlightPanel::timerCallback);
+    layout->addWidget(label);
+    setLayout(layout);
+    this->setFixedHeight(50);
+}
+
+// draws the stoplight
+void StartlightPanel::paintEvent(QPaintEvent *e) {
+    int radius = 21;
```

Comment:
I think this actually the diameter from what I can tell from the code

Reply:
Yeah you're correct, I should freshen up on my geometry